pkg/daemon/ceph/client/config.go

```diff
@@ -244,25 +244,25 @@ func addClientConfigFileSection(configFile *ini.File, clientName, keyringPath st
 
 // PopulateMonHostMembers extracts a list of just the monitor names, which will populate the "mon initial members"
 // and "mon hosts" global config field
-func PopulateMonHostMembers(monitors map[string]*MonInfo) ([]string, []string) {
-	monMembers := make([]string, len(monitors))
-	monHosts := make([]string, len(monitors))
+func PopulateMonHostMembers(clusterInfo *ClusterInfo) ([]string, []string) {
+	monMembers := make([]string, len(clusterInfo.Monitors))
+	monHosts := make([]string, len(clusterInfo.Monitors))
 
 	i := 0
-	for _, monitor := range monitors {
+	for _, monitor := range clusterInfo.Monitors {
 		monMembers[i] = monitor.Name
 		monIP := cephutil.GetIPFromEndpoint(monitor.Endpoint)
-
-		// This tries to detect the current port if the mon already exists
-		// This basically handles the transition between monitors running on 6790 to msgr2
-		// So whatever the previous monitor port was we keep it
-		currentMonPort := cephutil.GetPortFromEndpoint(monitor.Endpoint)
-
-		monPorts := [2]string{strconv.Itoa(int(Msgr2port)), strconv.Itoa(int(currentMonPort))}
-		msgr2Endpoint := net.JoinHostPort(monIP, monPorts[0])
-		msgr1Endpoint := net.JoinHostPort(monIP, monPorts[1])
-
-		monHosts[i] = "[v2:" + msgr2Endpoint + ",v1:" + msgr1Endpoint + "]"
+		if clusterInfo.RequireMsgr2 {
```

Comment:
Since we don't want to remove msgr1, I wonder if it won't be simpler to always have both? If so we won't need the `RequireMsgr2` method at all right?

Reply:
I'm thinking we need two different options:
- Allow both msgr1 and msgr2 in case clients don't support msgr2. This would still be the default since some clients may not yet support msgr2. But I need to dig deeper here to see if we really need this option.
- If connections are expected to be encrypted or compressed, we require all clients to be on msgr2 and disallow msgr1

Reply:
Since msgr2 requires a recent kernel 5.11, we will need to keep msgr1 as the default for a while.

Reply:
I think this makes sense to me. :+1: